eng/common/TestResources/New-TestResources.ps1

```diff
@@ -127,6 +127,38 @@ function Retry([scriptblock] $Action, [int] $Attempts = 5)
     }
 }
 
+# NewServicePrincipalWrapper creates an object from an AAD graph or Microsoft Graph service principal object type.
+# This is necessary to work around breaking changes introduced in Az version 7.0.0:
+# https://azure.microsoft.com/en-us/updates/update-your-apps-to-use-microsoft-graph-before-30-june-2022/
+function NewServicePrincipalWrapper($servicePrincipal)
+{
+    $spPassword = ""
+    if ($servicePrincipal.GetType().Name -eq 'MicrosoftGraphServicePrincipal') {
+        Write-Host "Creating password for service principal via MS Graph API"
```

Comment:
This feels like it may be mixing concerns;  we're wrapping the principal but also implicitly creating a new password.   Should we consider wrapping the principal creation and using this logic to unify the flow?

Reply:
Is this the same as your comment below? If so, yes I agree and will refactor.

Reply:
uh.... yeah.   Apparently, when you review over the course of a couple of hours, you forget what you already said.  😛